### PR TITLE
Add a validation for organization in the message context.

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
This PR fixes https://github.com/wso2-enterprise/choreo/issues/7348

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java [hunk 1]
```diff
@@ -1400,4 +1400,19 @@ public static boolean isDevPortalAnonymousEnabled(String tenantDomain) {
...
         return (String) ctx.get(RestApiConstants.ORGANIZATION);
     }
+
+    /**
+     * Method to extract the organization with validation
+     * @param ctx MessageContext
+     * @return organization
+     */
+
+    public static String getOrganizationWithValidation(MessageContext ctx) throws APIManagementException{
```
Comment (vlgunarathne) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java:1410]: What if we name the method `getValidatedOrganization()` ?

Comment (ShalkiWenushika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java]: Fixed via https://github.com/wso2/carbon-apimgt/pull/10726/commits/e665e6479c5d7b3b3ade8f735eb21615e02557f4

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 2]
```diff
@@ -780,8 +780,9 @@ public Response updateAPI(String apiId, APIDTO body, String ifMatch, MessageCont
                 (String[]) PhaseInterceptorChain.getCurrentMessage().getExchange()
                         .get(RestApiConstants.USER_REST_API_SCOPES);
         String username = RestApiCommonUtil.getLoggedInUsername();
-        String organization = RestApiUtil.getOrganization(messageContext);
+        // String organization = RestApiUtil.getOrganization(messageContext);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:783]: remove commented code

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 2 above)
Comment (ShalkiWenushika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: Fixed via https://github.com/wso2/carbon-apimgt/pull/10726/commits/037ecb03e0c0b05ce010a9ce42d1e43a1753c899

**Pull request merged**